spec/requests/permissions.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+describe '/projects/{project_id}/permissions' do
+  create_audio_recordings_hierarchy
+  it 'rejects an object that has both allow_anonymous and allow_logged_in set to true' do
+
+    # Set up a model
+    body = body_attributes_for(
+      :permission,
+      factory_args: {
+        user_id: nil,
+        level: :reader,
+        allow_anonymous: true,
+        allow_logged_in: true })
+
+    # Post to the server # some random token 1
+    Rails.logger.info("onwer token is #{owner_token}")
+    post "/projects/#{project.id}/permissions", params: body,**api_with_body_headers(owner_token)
+
+    # Assert the correct error is returned
+    expect_json_response
+    expect_error(:not_acceptable, 'You do not have sufficient permissions to access this page.', nil)
+
+  end
```

Comment:
Is this blocking a non-owner from editing the permissions for a project? Or blocking a project from setting both the `allow_anonymous` and `allow_logged_in` flags?

Reply:
The database is set up so that a single permission can be either for a single user, or for anonymous users, or for logged in users. You can't set more than one of those fields to nonhnull for any single permission, and this test is testing that